### ISSUE TITLE
get_many doesn't work with several backends

### ISSUE THREAD

**Issue by easysugar** [Krukov/cashews#73]
Seems that this method uses only one backend:
```python
# create different backends - for redis and memory
cache.setup('mem://', prefix='mem')
cache.setup('redis://', prefix='redis')

# set values
await cache.set('mem:123', 'memory')
await cache.set('redis:123', 'redis')

# call get_many

await cache.get_many('mem:123', 'redis:123')
# returns ('memory', None)
await cache.get_many('redis:123', 'mem:123')
# returns ('redis', None)
```

**Comment by AIGeneratedUsername** [comment]
There are two problems here. 

### The first problem is that `get_many` returns something wrong when we use Memory backend (but Redis backend works fine):
```python
import asyncio
from cashews import cache


async def func():
    cache.setup('mem://', prefix='foobar')

    print(await cache.set('foobar:123', 'value_1', expire=10_000))
    print(await cache.set('spameggs:123', 'value_2', expire=10_000))

    print(await cache.get_many('foobar:123', 'spameggs:123', 'foobar:123'))    # ('value_1', None, 'value_1')
    print(await cache.get_many('spameggs:123', 'foobar:123', 'spameggs:123'))  # ('value_2', None, 'value_2')


asyncio.run(func())
```
### Another problem is that using prefix like `"mem"` for Redis backend gives side effect that I can not explain
```python
import asyncio
from cashews import cache


async def func():
    cache.setup('redis://0.0.0.0', prefix='mem')

    print(await cache.set('mem:123', 'value_1', expire=1_000))
    print(await cache.set('mem:456', 'value_2', expire=1_000))

    print(await cache.get_many('mem:123', 'mem:456', 'mem:123'))  # (None, None, None)
    print(await cache.get_many('mem:456', 'mem:123', 'mem:456'))  # (None, None, None)


asyncio.run(func())
```

---

Let's wait for Krukov's help here. I never reviewed/investigated "cashews/wrapper.py" code.

**Comment by Krukov** [comment]
Hello, 
yes first issue that @easysugar  mentioned is a bug and it is a part of wrapper. I think I can fix it a little bit latter.

The second one is not depends on prefix and the bug is much more serious. That is bug because of `_` in value (`value_1`)
I added more details in separate issue

**Comment by Krukov** [comment]
So the second bug fixed https://github.com/Krukov/cashews/issues/74

**Comment by Krukov** [comment]
Fixed with 4.5.0
